cmd/readACH/main.go

```diff
@@ -56,22 +56,21 @@ func main() {
 		log.Panicf("Can not open file: %s: \n", err)
 	}
 
-	r := ach.NewReader(f)
-	achFile, err := r.Read()
+	achFile, err := ach.NewReader(f).Read()
 	if err != nil {
 		fmt.Printf("Issue reading file: %+v \n", err)
 	}
 
-	// ensure we have a validated file structure
-	if achFile.Validate(); err != nil {
-		fmt.Printf("Could not validate entire read file: %v", err)
-	}
-
 	// If you trust the file but it's formating is off building will probably resolve the malformed file.
 	if err := achFile.Create(); err != nil {
 		fmt.Printf("Could not create file with read properties: %v", err)
 	}
 
+	// ensure we have a validated file structure
+	if err := achFile.Validate(); err != nil {
```

Comment:
There was a subtle issue where `err` was not being redeclared.

Reply:
I'm surprised our linters aren't catching that.

Reply:
Yeah, I thought staticcheck caught stuff like this.